pkg/auth/condition.go

```diff
@@ -0,0 +1,212 @@
+	for fieldName, conditionValues := range fields {
+		// No field or empty value for the field - condition fails
+		contextValue, hasField := conditionCtx.Fields[fieldName]
+		if !hasField {
+			return false, nil
+		}
+		if contextValue == "" {
+			return false, nil
+		}
+
+		// Parse the context value as an IP address
+		contextIP := net.ParseIP(contextValue)
+		if contextIP == nil {
+			return false, fmt.Errorf("invalid IP format in field %s: %s", fieldName, contextValue)
+		}
+
+		// Check if context IP matches any of the condition values for this field (OR logic within field)
+		fieldMatched := false
+		for _, value := range conditionValues {
+			value = strings.TrimSpace(value)
+
+			// First try to parse as CIDR
+			if _, ipNet, err := net.ParseCIDR(value); err == nil {
+				if ipNet.Contains(contextIP) {
+					fieldMatched = true
+					break
+				}
+				continue
+			}
+
+			// Try to parse as a single IP address
+			if singleIP := net.ParseIP(value); singleIP != nil {
+				if contextIP.Equal(singleIP) {
+					fieldMatched = true
+					break
+				}
+				continue
+			}
+
+			// Invalid format
+			return false, fmt.Errorf("invalid IP/CIDR format in field %s: %s", fieldName, value)
+		}
+
+		// If this field didn't match any values, entire condition fails (AND logic between fields)
+		if !fieldMatched {
+			return false, nil
+		}
+	}
+
+	// All fields matched
+	return true, nil
+}
+
+// OperatorFactory returns the appropriate operator for a given operator name
+func OperatorFactory(operatorName string) (ConditionOperator, error) {
+	switch operatorName {
+	case OperatorNameIpAddress:
+		return &IpAddressOperator{}, nil
+	default:
+		return nil, fmt.Errorf("unsupported condition operator: %s", operatorName)
+	}
+}
+
+// EvaluateConditions checks if all conditions in the map are satisfied
+// conditions is a map where keys are operator names and values are maps of field names to value arrays
+// AWS IAM format: {"IpAddress": {"SourceIp": ["203.0.113.0/24", "198.51.100.25/32"]}}
+// Returns true only if all conditions pass (AND logic)
+func EvaluateConditions(conditions map[string]map[string][]string, conditionCtx *ConditionContext) (bool, error) {
+	if len(conditions) == 0 {
+		// No conditions mean always pass
+		return true, nil
+	}
+
+	// Validate context
+	if conditionCtx == nil {
+		return false, fmt.Errorf("invalid condition context: context is nil")
+	}
+
+	// All conditions must pass (AND logic)
+	for operatorName, fields := range conditions {
+		operator, err := OperatorFactory(operatorName)
+		if err != nil {
+			return false, err
+		}
+
+		passed, err := operator.Evaluate(fields, conditionCtx)
+		if err != nil {
+			return false, err
+		}
+
+		if !passed {
+			// One condition failed, entire statement fails
+			return false, nil
+		}
+	}
+
+	// All conditions passed
+	return true, nil
+}
+
+// ExtractClientIP extracts the client IP from the given headers
+// Checks X-Forwarded-For header first (for proxy/load balancer scenarios)
```

Comment:
Great thing you X-Forwarded-For header first! I was about to ask about this.
I'm not sure if this method is relevant for the auth pkg, but if we decide to keep it here, let's use it to add a debug log that gives the user a hint about which IP we resolved (from `X-Forwarded` header, `X-Real-IP`, etc...). I believe it will be helpful in many cases.

Reply:
1. Can move this function to `httputil` package it can be outside auth, wdyt?
2. Added log of which fields on context while we perform permission check.

Reply:
1. extract client ip moved to httputil package
2. added log level in debug - usually I prefer not to debug these issues by logging at the lowest level. I also added logging in case a condition fails and it will print the extracted ip used in the permission checks.